Xamarin.Forms.Core/BindableObject.cs

```diff
@@ -486,7 +486,7 @@ BindablePropertyContext GetContext(BindableProperty property)
 			for (var i = 0; i < properties.Count; i++)
 			{
 				BindablePropertyContext context = properties[i];
-				if (ReferenceEquals(context.Property, property))
+				if (context != null && ReferenceEquals(context.Property, property))
```

Comment:
@rotorgames if I look at https://bugzilla.xamarin.com/show_bug.cgi?id=23940 description, the problem arise when you modify a BindableProperty from another thread than the main one.

Is this the case for you ?